tools/codegen/api/python.py

```diff
@@ -271,22 +389,36 @@ class PythonSignature:
     # method or function signature?
     method: bool
 
+    # is this a pyi signature?
+    # Note: the python signatures used in python bindings vs. pyi have the same state, but
+    # are displayed differently. Pyi signatures has different syntax for arguments,
+    # and don't display return types
+    pyi: bool
+
     @property
     def deprecated(self) -> bool:
         return False
 
     def arguments(
-        self, *, skip_outputs: bool = False, skip_tensor_options: bool = False
+            self, *, skip_outputs: bool = False, skip_tensor_options: bool = False, hacky_add_output: bool = False
     ) -> Tuple[Union[PythonArgument, PythonOutArgument], ...]:
         result: List[Union[PythonArgument, PythonOutArgument]] = []
         result.extend(self.input_args)
         result.extend(self.input_kwargs)
         if self.output_args is not None and not skip_outputs:
             result.append(self.output_args)
+        # TODO: remove HACK
```

Comment:
As an example, if I take this out, this is what happens to `var` in `_VF.pyi`:

Before
```
 @overload
 def var(input: Tensor, unbiased: _bool=True, *, out: Optional[Tensor]=None) -> Tensor: ...
 @overload
 def var(input: Tensor, dim: Union[_int, _size], unbiased: _bool=True, keepdim: _bool=False, *, out: Optional[Tensor]=None) -> Tensor: ...
```

After
```
 @overload
 def var(input: Tensor, unbiased: _bool=True) -> Tensor: ...
 @overload
 def var(input: Tensor, dim: Union[_int, _size], unbiased: _bool=True, keepdim: _bool=False, *, out: Optional[Tensor]=None) -> Tensor: ...
```

The first `var` signature no longer has an optional `out` arg. That seems correct, since the corresponding listing in native_functions.yaml doesn't have an `out` variant, but the second overload does: https://github.com/pytorch/pytorch/blob/93973ee6993c97c8b9d60c4f720423bc625073ea/aten/src/ATen/native/native_functions.yaml#L4103

Reply:
In situations like this, you can always check your intuition by loading up PyTorch and seeing if `torch.var(input, unbiased, out=out)` works or not.

Reply:
yep good point - and that breaks. So pyi is definitely wrong here